lib/Alchemy/Phrasea/Controller/Prod/EditController.php

```diff
@@ -375,4 +378,144 @@ public function applyAction(Request $request) {
 
         return $this->app->json(['success' => true]);
     }
+
+
+    /**
+     * route GET "../prod/records/edit/presets/{preset_id}"
+     *
+     * @param Request $request
+     * @param $preset_id
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function presetsLoadAction($preset_id)
+    {
+        $ret = [];
+
+        if ( ($preset = $this->getPresetRepository()->find($preset_id)) === null) {
+            $this->app->abort(404, sprintf("Preset with id '%' could not be found", $preset_id));
+        }
+
+        $fields = [];
+        foreach ($preset->getData() as $field) {
+            $fields[$field['name']] = $field['value'];
+        }
+
+        $ret['fields'] = $fields;
+
+        return $this->app->json($ret);
+    }
+
+    /**
+     * route GET "../prod/records/edit/presets"
+     *
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function presetsListAction(Request $request)
+    {
+        $sbas_id = $request->get('sbas_id');
+        $user = $this->app['authentication']->getUser();
+
+        $ret = [];
+
+        $ret['html'] = $this->getPresetHTMLList($sbas_id, $user);
+
+        return $this->app->json($ret);
+    }
+
+    /**
+     * route DELETE "../prod/records/edit/presets/{preset_id}"
+     *
+     * @param Request $request
+     * @param $preset_id
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function presetsDeleteAction($preset_id)
+    {
+        $user = $this->app['authentication']->getUser();
+
+        $ret = [];
+
+        /** @var Preset $preset */
+        if ( ($preset = $this->getPresetRepository()->find($preset_id)) === null) {
```

Comment:
This is done many times in the controller. Please Factorize it into `findOr404` or similar method.